test/smoke/dashCore.js

```diff
@@ -1,18 +1,115 @@
+
+  // The number of blocks should be almost the same (-3/+3) and block hash of particular
+  // block height should be the same. Using `blocks`
+  // and `bestblockhash` from `GetBlockChainInfo`.
+  // Verify it on masternodes, wallet-nodes, full-nodes, miners
+  it('should have correct blockhash and blocks count', async () => {
+    const blockCounts = [];
+    const blockhashes = {};
+    let blockchainInfo;
+    for (let i = 0; i < nodeNames.length; i++) {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeNames[i]].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+      const blockCount = await rpc.getBlockCount();
+      blockchainInfo = await rpc.getBlockchainInfo();
+      blockCounts.push(blockCount.result);
+      if (!blockhashes[blockchainInfo.result.blocks]) {
+        blockhashes[blockchainInfo.result.blocks] = blockchainInfo.result.bestblockhash;
+      }
+    }
+    expect(blockhashes[blockchainInfo.result.blocks]).to.be.equal(
+      blockchainInfo.result.bestblockhash,
+    );
+    expect(Math.max(...blockCounts) - Math.min(...blockCounts)).to.be.below(3);
   });
+});
+
+describe('Masternodes', () => {
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    describe(nodeName, () => {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+      // masternode status
+      it('should masternodes be enabled', async () => {
+        const masternodelist = await rpc.masternodelist();
+        console.log(masternodelist);
+        const idNodes = Object.keys(masternodelist.result);
+        const masterIps = [];
+        for (let i = 0; i < idNodes.length; i++) {
+          masterIps.push(masternodelist.result[idNodes[i]].address.split(':')[0]);
+          expect(masternodelist.result[idNodes[i]].status).to.be.equal('ENABLED');
+        }
+        const masterIpsInv = [];
+        for (const nName of networkConfig.inventory.masternodes.hosts) {
+          masterIpsInv.push(networkConfig.inventory._meta.hostvars[nName].public_ip);
+        }
+        expect(masterIps.sort()).to.deep.equal(masterIpsInv.sort());
+      });
+    });
+  });
+
   describe('Miners', () => {
     // waitfornewblock
-    it('should mine blocks');
+    it('should mine blocks', async function () {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[
+          networkConfig.inventory.masternodes.hosts[0]].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+      this.timeout(160000);
+      const blockCount = await rpc.getBlockCount();
```

Comment:
waitfornewblock

Reply:
changed

Reply:
We need just `await rpc.waitForNewBlock(1);` here.

Reply:
await rpc.waitForNewBlock(1) doesn't work now.
for example the following code
``` 
     const { result } = await rpc.getBlockCount();
      const { result: { height } } = await rpc.waitForNewBlock(1);
      expect(result + 1).to.be.equal(height);
```
doesn't wait new block. and rpc.waitForNewBlock returns height=result imediately